dotCMS/src/main/java/com/dotcms/publisher/util/DependencySet.java

```diff
@@ -140,7 +184,13 @@ private boolean addOrClean ( String assetId, Date assetModDate, Boolean cleanFor
 				
 				if(modifiedOnCurrentEnv) {
 					try {
-                        asset = new PushedAsset(bundleId, assetId, assetType, new Date(), env.getId());
+						if(endpointIds != null && publisher != null){
```

Comment:
If the "endpointIds" and "publisher" variables are null, we can just have one single statement. We don't need to IF clause.